Integrations/DigitalDefense/DigitalDefense.py

```diff
@@ -0,0 +1,596 @@
+    else:
+        scan['name'] = ("Demisto Scan")
+    scan['description'] = "New network device auto scan launch from Demisto."
+    scan['schedule'] = {
+        "id": None,
+        "end_date": None,
+        "start_date":now.strftime("%Y-%m-%dT%H:%M:%S%z"),
+        "recurring": False,
+        "recurrence_rules": [],
+        "timezone": tz,
+        "timezone_offset": tzoffset
+    }
+    scan['workflow'] = "va_workflow"
+    scan['exclude_from_active_view'] = False
+    scan['notify'] = False
+    scan['internal'] = True
+    scan['recipients'] = []
+    scan['scan_policy'] = scan_policy
+    scan['scan_speed'] = "normal"
+    scan['asset_groups'] = []
+    scan['asset_filter'] = {}
+
+    business_group = get_business_group()
+    if business_group:
+        scan['businessgroups'] = [{"id": business_group['id']}]
+    scan['adhoc_targets'] = []
+    scan['adhoc_targets'].append({
+        "rule_action": "include",
+        "network_profile_id": int(scanner_data['profile_id']),
+        "inclusion": "full",
+        "ip_address_range": {
+            "low_ip_address": low_ip_address,
+            "high_ip_address": high_ip_address,
+            "low_ip_number": low_ip_number,
+            "high_ip_number": high_ip_number,
+            "ipv6": False,
+            "dhcp": False,
+            "fragile": False,
+            "cidr_block": None
+        }
+    })
+    return scan
+
+
+def scan_asset(ip_address, scan_policy):
+    if "-" in ip_address:
+        low_ip_address = ip_address.split("-")[0].strip()
+        high_ip_address = ip_address.split("-")[1].strip()
+    else:
+        low_ip_address = ip_address
+        high_ip_address = ip_address
+    scan_payload = build_scan(low_ip_address, high_ip_address, scan_policy)
+    header = {}
+    header['Authorization'] = 'Token ' + str(API_TOKEN)
+    header['Content-Type'] = "application/json;charset=utf-8"
+    resp = requests.post(SCAN_ENDPOINT, data=json.dumps(scan_payload), headers=header)
+    resp.raise_for_status()
+    scan_response = json.loads(resp.text)
+    return scan_response
+
+
+def scan_policy_exists(policy_selected):
+    policy_url = SCAN_ENDPOINT + "policies/"
+    demisto.debug("FrontlineVM scan_policy_exists -- checking if user defined policy exists within Frontline.Cloud")
+    try:
+        resp = requests.get(policy_url, headers={'Authorization': 'Token ' + str(API_TOKEN)})
+        resp.raise_for_status()
+        data = json.loads(resp.text)
+        for policy in data:
+            if str(policy_selected) == policy.get('name', ""):
+                return True
+        return False
+    except Exception as err:
+        demisto.error("FrontlineVM scan_policy_exists -- Exception: " + str(err.message))
+        return_error(err.message)
+
+
+def scan_asset_command():
+    ip_address = demisto.args().get('ip_address')
+    scan_policy = str(demisto.args().get('scan_policy'))
+    if scan_policy != "Default" and not scan_policy_exists(scan_policy):
```

Comment:
The purpose of the condition `scan_policy != "Default" ` is because we already know the default policy name 'Default' exists for every user, if not "Default" then check if scan policy exists.

Changed to check if policy name exists no matter what the name is, even if 'Default' for easier comprehension.

Reply:
`scan_policy` is in fact simply a policy name. Although our product defines this as **Scan Policy** and our clients are familiar with this terminology, I've kept the argument name as `scan_policy` but have changed the variable name to `policy_name` for better code comprehension.